cmd/insprd/auth/jwt/auth.go

```diff
@@ -13,56 +15,82 @@ import (
 	"github.com/inspr/inspr/pkg/auth/models"
 	"github.com/inspr/inspr/pkg/ierrors"
 	"github.com/inspr/inspr/pkg/rest/request"
+	"github.com/lestrrat-go/jwx/jwa"
 	"github.com/lestrrat-go/jwx/jwt"
 )
 
 // JWTauth implements the Auth interface for jwt authetication provider
 type JWTauth struct {
-	authURL string
+	PublicKey *rsa.PublicKey
+	authURL   string
 }
 
-// NewJWTauth returns a JWTAuth object
-func NewJWTauth() *JWTauth {
+// NewJWTauth takes an *rsa.PublicKey and returns an
+// structure that implements the auth interface
+func NewJWTauth(rsaPublicKey *rsa.PublicKey) *JWTauth {
 	url, ok := os.LookupEnv("AUTH_PATH")
 	if !ok {
 		panic("AUTH_PATH not found")
 	}
 	return &JWTauth{
-		authURL: url,
+		PublicKey: rsaPublicKey,
+		authURL:   url,
 	}
 }
 
-//Validate receives the
-func (JA *JWTauth) Validate(token []byte) (models.Payload, []byte, error) {
-	jwtToken, err := jwt.Parse(token, jwt.WithValidate(true))
+// Validade is a wrapper that checks the token of the http request and if it's
+// valid, proceeds to execute the request and if it isn't valid returns an error
+func (JA *JWTauth) Validate(token []byte) (*models.Payload, []byte, error) {
+
+	jwtToken, err := jwt.Parse(
+		token,
+		jwt.WithValidate(true),
+		jwt.WithVerify(jwa.RS256, JA.PublicKey),
+	)
+
 	// not valid token
 	if err != nil {
-		return models.Payload{}, token, err
+		return &models.Payload{}, token, err
 	}
 
 	expiration, found := jwtToken.Get(jwt.ExpirationKey)
 	now := time.Now()
 
 	// expired
-	if now.After(expiration.(time.Time)) || !found {
+	if !found || now.After(expiration.(time.Time)) {
 		newToken, err := JA.Refresh(token)
 		if err != nil {
-			return models.Payload{},
+			return &models.Payload{},
 				token,
-				errors.New("error refreshing token")
+				ierrors.
+					NewError().
+					InternalServer().
+					Message("error refreshing token").
+					Build()
 		}
 		token = newToken
 	}
 
 	// gets payload from token
-	payload, found := jwtToken.Get("payload")
+	payloadData, found := jwtToken.Get("payload")
 	if !found {
-		return models.Payload{},
+		return &models.Payload{},
 			token,
 			errors.New("payload not found")
 	}
 
-	return payload.(models.Payload), token, nil
+	jwtJSON, err := json.Marshal(payloadData)
+	if err != nil {
+		return &models.Payload{}, token, err
+	}
+
+	var payload models.Payload
+	err = json.Unmarshal(jwtJSON, &payload)
+	if err != nil {
+		return &models.Payload{}, token, err
+	}
+
+	return &payload, token, nil
```

Comment:
Because the function expected return is a pointer to a structure, you can just returl 'nil' when there's an error